Source/GRChomboCore/ChomboParameters.hpp

```diff
@@ -281,6 +291,77 @@ class ChomboParameters
         pout() << endl;
     }
 
+    void check_params()
+    {
+        check_parameter("L", L, L > 0.0, "must be > 0.0");
+        check_parameter("max_level", max_level, max_level >= 0, "must be >= 0");
+        // the following check assumes you will be taking some fourth (or
+        // higher) order one-sided derivatives
+        check_parameter("num_ghosts", num_ghosts, num_ghosts >= 3,
+                        "must be >= 3");
+        check_parameter("tag_buffer_size", tag_buffer_size,
+                        tag_buffer_size >= 0, "must be >= 0");
+        check_parameter("dt_multiplier", dt_multiplier, dt_multiplier > 0.0,
+                        "must be > 0.0");
+        check_parameter("max_grid_size/max_box_size", max_grid_size,
+                        max_grid_size >= 0, "must be >= 0");
+        check_parameter("block_factor/min_box_size", block_factor,
+                        block_factor >= 1, "must be >= 1");
+        // note that this also enforces block_factor <= max_grid_size
+        // if max_grid_size > 0
+        check_parameter("block_factor/min_box_size", block_factor,
+                        max_grid_size % block_factor == 0,
+                        "must divide max_grid_size/max_box_size = " +
+                            std::to_string(max_grid_size));
+        FOR1(idir)
+        {
+            std::string Ni_string = "N" + std::to_string(idir + 1);
+            std::string invalid_message = "must divide " + Ni_string;
+            if (boundary_params.reflective_boundaries_exist)
+            {
+                invalid_message += " (or " + Ni_string + "_full/2)";
+            }
+            invalid_message += " = " + std::to_string(ivN[idir] + 1);
+            check_parameter("block_factor/min_box_size", block_factor,
+                            (ivN[idir] + 1) % block_factor == 0,
+                            invalid_message);
+        }
+        check_parameter("fill_ratio", fill_ratio,
+                        (fill_ratio > 0.0) && (fill_ratio <= 1.0),
+                        "must be > 0 and <= 1");
+        // (MR); while this would technically work (any plot files would just
+        // overwrite a checkpoint file), I think a user would only ever do
+        // this unintentinally
```

Comment:
Ah, actually I would make it a check. I did this once and I think rather than simply overwriting, everything gets screwed up in the plot file. Also it means you won't have checkpoints as you expect which could be painful.

Reply:
OK, done